### PR TITLE
Fundamental changes for NN framework

### PR DESCRIPTION
**Pull request by c-yuwei**
This PR improves the generalizability of NN framework, making the pipelines more flexible to various types of input data (e.g., graph measures, adj matrix, multiplex, multigraph). 
The main implementation is introducing NNGroup and NNSubject to coordinate the input and output for NN.
By doing so, any input from ST, CON, FUN, multigraph, or multiplex can be altered and adapted to desired NN model.

The following elements have been tested:
- [x] 'test_NNBase'
- [x] 'test_NNData'
- [x] 'test_NNEvaluator'
- [x] 'test_NNGroup'
- [x] 'test_NNSubject'
- [x] 'test_NNClassifierData'
- [x] 'test_NNClassifierEvaluator'
- [x] 'test_NNClassifierDNN'
- [x] 'test_NNRegressorData'
- [x] 'test_NNRegressorEvaluator'
- [x] 'test_NNRegressorDNN'
- [x] 'test_NNClassifierData_CON_WU'
- [x] 'test_NNRegressorData_CON_WU'

### CODE REVIEW

File: braph2genesis/src/nn/_NNGroup.gen.m [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+Element, Subject, IndexedDictionary
+
+
+%% ¡props!
+
+%%% ¡prop!
+FEATURE_LABEL (metadata, string) is an feature label for feature mask.
+
+%%% ¡prop!
+FEATURE_MASK (metadata, cell) is an feature mask for all the NN subjects in the group.
```
Comment (giovannivolpe) [inline braph2genesis/src/nn/_NNGroup.gen.m]: these should be data

File: braph2genesis/src/nn/_NNGroup.gen.m [hunk 2]
```diff
@@ -0,0 +1,20 @@
+%% ¡header!
+NNGroup < Group (gr, group of subjects) is a group of subjects.
+
+%%% ¡description!
+NNGroup represents a group of NNSubjects whose class is defined in the property SUB_CLASS.
+Group provides the methods necessary to handle groups of subjects.
```
Comment (c-yuwei) [inline braph2genesis/src/nn/_NNGroup.gen.m:6]: describe here what is the feature mask

File: braph2genesis/src/nn/_NNSubject.gen.m [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+%%% ¡prop!
+INPUT (data, cell) is the input obtained from the data of subject for the neural network analysis.
+% % %%%% ¡gui!
+% % pr = PPSubjectCON_CON('EL', sub, 'PROP', SubjectCON.CON, varargin{:})
+
+%%% ¡prop!
+INPUT_FS (data, cell) is the input with feature selection.
+% % %%%% ¡gui!
+% % pr = PPSubjectCON_CON('EL', sub, 'PROP', SubjectCON.CON, varargin{:});
```
Comment (giovannivolpe) [inline braph2genesis/src/nn/_NNSubject.gen.m:25]: propos: input (data), mask (data), input_masked (result)

File: braph2genesis/src/nn/_NNGroup.gen.m [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+It manages the subjects as an indexed dictionary of NNSubjects SUB_DICT, 
+whose methods can be used to inspect, add or remove subjects.
+
+%%% ¡seealso!
+Element, Subject, IndexedDictionary
+
+%% ¡props!
+
+%%% ¡prop!
+FEATURE_MASK (metadata, cell) is an feature mask for all the NN subjects in the group.
```
Comment (giovannivolpe) [inline braph2genesis/src/nn/_NNGroup.gen.m:16]: no mask, but inputs and inputs_masked as results that collect the data from subjects

File: braph2genesis/src/nn/_NNSubject.gen.m [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+%% ¡props!
+
+%%% ¡prop!
+G (data, item) is a graph obtained from the data of the subject.
+%%%% ¡settings!
+'Graph'
+
+%%% ¡prop!
+INPUT (data, cell) is the input obtained from the data of subject for the neural network analysis.
```
Comment (giovannivolpe) [inline braph2genesis/src/nn/_NNSubject.gen.m:18]: make it into cell as we discussed

File: braph2genesis/src/nn/_NNSubject.gen.m [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+% % %%%% ¡gui!
+% % pr = PPSubjectCON_CON('EL', sub, 'PROP', SubjectCON.CON, varargin{:})
+
+%%% ¡prop!
+INPUT_FS (data, cell) is the input with feature selection.
+% % %%%% ¡gui!
+% % pr = PPSubjectCON_CON('EL', sub, 'PROP', SubjectCON.CON, varargin{:});
+
+%%% ¡prop!
+TARGET (data, string) is the target of the subject for the neural network analysis.
```
Comment (giovannivolpe) [inline braph2genesis/src/nn/_NNSubject.gen.m:28]: cell

**Review by giovannivolpe — CHANGES_REQUESTED** [review]

**Comment by c-yuwei** [comment]
The following elements have been tested:
- [x] 'test_NNBase'
- [x] 'test_NNData'
- [x] 'test_NNEvaluator'
- [x] 'test_NNGroup'
- [x] 'test_NNSubject'
- [x] 'test_NNClassifierData'
- [x] 'test_NNClassifierEvaluator'
- [x] 'test_NNClassifierDNN'
- [x] 'test_NNClassifierData_CON_WU'

**Pull request merged**